rasterio/_io.pyx

```diff
@@ -1479,6 +1479,8 @@ cdef class RasterUpdater(RasterReader):
 
         # Normally, we expect a CRS dict.
         if isinstance(crs, dict):
+            crs = CRS(crs)
+        if isinstance(crs, CRS):
```

Comment:
@jameshiebert Should the order of 1481 and 1483 be flipped?  Since we now prefer `CRS` objects, I think we should test that first.